### PR TITLE
Overhaul DiscGuard detection

### PR DESCRIPTION
**Pull request by SilasLaspada**
* Verify some existing checks for DiscGuard.
* Add several new executable and path checks.
* Add note for remaining unconfirmed check.
* Add notes and research in comments.
* Add related protection "DiscAudit" to list of protections that needs to be added.

### CODE REVIEW

File: BurnOutSharp/ProtectionType/DiscGuard.cs [hunk 1]
```diff
@@ -1,23 +1,168 @@
...
+
+            // Get the .vbn, if it exists
+            var DiscGuardSection = pex.ReadRawSection(".vbn");
+            if (DiscGuardSection != null)
+            {
+                var matchers = new List<ContentMatchSet>
+                {
+                    // Found in "T29.dll" (Redump entry 31914).
+                    // This check should be as long as the following check, as this data is nearly identical (including length) in the original files, but for some reason the section ends early, causing part of the remaining data to not be part of a section.
+                    new ContentMatchSet(new byte?[]
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/DiscGuard.cs:49]: If this is a subset of the other check, is the other check needed? We have a few places where the longer check is commented out until we can prove it's needed.

File: BurnOutSharp/ProtectionType/DiscGuard.cs [hunk 2]
```diff
@@ -1,23 +1,168 @@
...
+                    }, "DiscGuard"),
+                };
+
+                string match = MatchUtil.GetFirstMatch(file, DiscGuardSection, matchers, includeDebug);
+                if (!string.IsNullOrWhiteSpace(match))
+                    return match;
+            }
+
+            // Get the .rsrc, if it exists
+            var ResourceSection = pex.ReadRawSection(".rsrc");
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/DiscGuard.cs:100]: Resource section should also be one of the pre-read sections. This _will_ be automatically used, but you can use it directly as well.

```suggestion
            var rsrcSection = pex.ReadRawSection(".rsrc");
```

File: BurnOutSharp/ProtectionType/DiscGuard.cs [hunk 3]
```diff
@@ -1,23 +1,168 @@
...
+                    }, "DiscGuard"),
+                };
+
+                string match = MatchUtil.GetFirstMatch(file, ResourceSection, matchers, includeDebug);
+                if (!string.IsNullOrWhiteSpace(match))
+                    return match;
+            }
+
+            // Found in "IOSLinksys.dll" (Redump entries 31914, 46743, 46961, 79284, and 79374).
+            string name = pex.FileDescription;
```
Comment (mnadareski) [inline BurnOutSharp/ProtectionType/DiscGuard.cs:124]: If these values are just as reliable, move them to the top because they are far less intensive.

**Review by mnadareski — CHANGES_REQUESTED** [review]

File: BurnOutSharp/ProtectionType/DiscGuard.cs (hunk 1 above)
Comment (SilasLaspada) [reply BurnOutSharp/ProtectionType/DiscGuard.cs:49]: It is not a subset of the other check, they are subtly different from one another. The fifth byte in the sequence is the first differing one.

**Review by mnadareski — APPROVED** [review]

**Pull request merged**